src/main/java/de/tum/in/www1/artemis/service/metis/PostService.java

```diff
@@ -529,4 +529,49 @@ private void mayInteractWithPostElseThrow(Post post, User user, Course course) {
             authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, user);
         }
     }
+
+    /**
+     * helper method that fetches groups and authorities of all posting authors in a list of Posts
+     * @param postsInCourse list of posts whose authors are populated with their groups, authorities, and authorRole
+     */
+    private void setAuthorRoleOfPostings(List<Post> postsInCourse) {
+        // prepares a unique set of userIds that authored the current list of postings
+        Set<Long> userIds = new HashSet<>();
+        postsInCourse.forEach(post -> {
+            userIds.add(post.getAuthor().getId());
+            post.getAnswers().forEach(answerPost -> userIds.add(answerPost.getAuthor().getId()));
+        });
+
+        // fetches and sets groups and authorities of all posting authors involved, which are used to display author role icon in the posting header
+        // converts fetched set to hashmap type for performant matching of authors
+        Map<Long, User> authors = userRepository.findAllWithGroupsAndAuthoritiesByIdIn(userIds).stream().collect(Collectors.toMap(user -> user.getId(), Function.identity()));
+
+        // sets respective author role to display user authority icon on posting headers
+        postsInCourse.forEach(post -> {
+            post.setAuthor(authors.get(post.getAuthor().getId()));
+            setAuthorRoleForPosting(post, post.getCoursePostingBelongsTo());
+            post.getAnswers().forEach(answerPost -> {
+                answerPost.setAuthor(authors.get(answerPost.getAuthor().getId()));
+                setAuthorRoleForPosting(answerPost, answerPost.getCoursePostingBelongsTo());
+            });
+        });
+    }
+
+    /**
+     * helper method that assigns authorRoles of postings in accordance to user groups and authorities
+     * @param posting       posting to assign authorRole
+     * @param postingCourse course that the post belongs to, must be explicitly fetched and provided to handle new post creation case
+     */
+    private void setAuthorRoleForPosting(Posting posting, Course postingCourse) {
+        if (authorizationCheckService.isAtLeastInstructorInCourse(postingCourse, posting.getAuthor())) {
+            posting.setAuthorRole(UserRole.INSTRUCTOR);
+        }
+        else if (authorizationCheckService.isTeachingAssistantInCourse(postingCourse, posting.getAuthor())
+                || authorizationCheckService.isEditorInCourse(postingCourse, posting.getAuthor())) {
```

Comment:
```suggestion
        else if (authorizationCheckService.isTeachingAssistantInCourse(postingCourse, posting.getAuthor())) {
```
An editor is also at least a teaching assistant

Reply:
please see my previous reply
https://github.com/ls1intum/Artemis/pull/5473#discussion_r930762334